src/helpers/helpers.dom.js

```diff
@@ -34,112 +29,113 @@ function parseMaxStyle(styleValue, node, parentProperty) {
+	} else {
+		const rect = canvas.getBoundingClientRect();
+		x = source.clientX - rect.left;
+		y = source.clientY - rect.top;
 	}
-
-	return calculateRelativePositionFromClientXY(source, chart);
+	return {x, y};
 }
 
-function calculateRelativePositionFromClientXY(source, chart) {
-	const {clientX: x, clientY: y} = source;
-
-	const canvasElement = chart.canvas;
-	const devicePixelRatio = chart.currentDevicePixelRatio;
-	const boundingRect = canvasElement.getBoundingClientRect();
-	// Scale mouse coordinates into canvas coordinates
-	// by following the pattern laid out by 'jerryj' in the comments of
-	// https://www.html5canvastutorials.com/advanced/html5-canvas-mouse-coordinates/
-	const paddingLeft = parseFloat(getStyle(canvasElement, 'padding-left'));
-	const paddingTop = parseFloat(getStyle(canvasElement, 'padding-top'));
-	const paddingRight = parseFloat(getStyle(canvasElement, 'padding-right'));
-	const paddingBottom = parseFloat(getStyle(canvasElement, 'padding-bottom'));
-	const width = boundingRect.right - boundingRect.left - paddingLeft - paddingRight;
-	const height = boundingRect.bottom - boundingRect.top - paddingTop - paddingBottom;
-
-	// We divide by the current device pixel ratio, because the canvas is scaled up by that amount in each direction. However
-	// the backend model is in unscaled coordinates. Since we are going to deal with our model coordinates, we go back here
+export function getRelativePosition(evt, chart) {
+	const {canvas, currentDevicePixelRatio} = chart;
+	const style = getComputedStyle(canvas);
+	const borderBox = style.boxSizing === 'border-box';
+	const paddings = getPositionedStyle(style, 'padding');
+	const {x, y} = getCanvasPosition(evt, canvas);
+
+	let {width, height} = chart;
+	if (borderBox) {
+		const borders = getPositionedStyle(style, 'border');
+		width -= paddings.width + borders.width;
+		height -= paddings.height + borders.height;
+	}
 	return {
-		x: Math.round((x - boundingRect.left - paddingLeft) / (width) * canvasElement.width / devicePixelRatio),
-		y: Math.round((y - boundingRect.top - paddingTop) / (height) * canvasElement.height / devicePixelRatio)
+		x: Math.round((x - paddings.left) / width * canvas.width / currentDevicePixelRatio),
+		y: Math.round((y - paddings.right) / height * canvas.height / currentDevicePixelRatio)
 	};
 }
 
-function fallbackIfNotValid(measure, fallback) {
-	return typeof measure === 'number' ? measure : fallback;
+const infinity = Number.POSITIVE_INFINITY;
+
+function getContainerSize(canvas, width, height) {
+	let maxWidth, maxHeight;
+
+	if (width === undefined || height === undefined) {
+		const container = _getParentNode(canvas);
+		if (!container) {
+			width = canvas.clientWidth;
+			height = canvas.clientHeight;
+		} else {
+			const rect = container.getBoundingClientRect(); // this is the border box of the container
+			const containerStyle = getComputedStyle(container);
+			const containerBorder = getPositionedStyle(containerStyle, 'border');
```

Comment:
Does this need a check for the box sizing of the container?

Reply:
According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect):

> The element's size is equal to its width/height + padding + border-width in the case that the standard box model is being used, or width/height only if box-sizing: border-box has been set on it.

So its always the `border-box` that is returned.